src/graphcast_agent/message_typing.rs

```diff
@@ -340,13 +349,27 @@ impl<T: Message + ethers::types::transaction::eip712::Eip712 + Default + Clone +
 pub async fn check_message_validity<
     T: Message + ethers::types::transaction::eip712::Eip712 + Default + Clone + 'static,
 >(
-    graphcast_message: GraphcastMessage<T>,
+    mut graphcast_message: GraphcastMessage<T>,
     nonces: &Arc<Mutex<NoncesMap>>,
     registry_subgraph: &str,
     network_subgraph: &str,
     graph_node_endpoint: &str,
     local_graphcast_id: String,
 ) -> Result<GraphcastMessage<T>, BuildMessageError> {
+    // These mocks are only used in tests and are temporary
+    // Will be removed once tests are ported to use cfg(test)
+    if env::var("CHECK").is_ok() {
+        if let Ok(mock_nonce) = env::var("MOCK_NONCE") {
+            debug!("Mock nonce: {}", mock_nonce);
+            graphcast_message.nonce = mock_nonce.parse().expect("Not a valid number");
```

Comment:
I think we spoke about how the mocked messages can be filled with these mocked values (nonce and hash) before sending?